pkg/auth/inject.go

```diff
@@ -57,6 +58,16 @@ func (m DependencyMap) Provide(dependencyName string, r *http.Request) interface
 			// case "skygear":
 			// 	return XXX
 		}
+	case "AuditTrail":
+		tConfig := config.GetTenantConfig(r)
+		if tConfig.UserAudit.Enabled && tConfig.UserAudit.TrailHandlerURL != "" {
+			trail, err := coreAudit.NewTrail(tConfig.UserAudit.TrailHandlerURL, r)
+			if err != nil {
+				panic(err)
+			}
+			return trail
+		}
+		return nil
```

Comment:
Always return an `trail`, just config it according to the TenantConfig. Take reference to the `InitTrailHandler`.